src/modules/types/user.ts

```diff
@@ -1,6 +1,19 @@
+import { TimeString } from "./route";
+
+export type UserId = number;
+export type Name = string;
+export type Email = string;
+export type GoogleId = string;
+export type PictureUrl = string;
+
 export type User = {
-  name: string;
-  avatarSrc?: URL;
+  id: UserId;
```

Comment:
The type can just be `string`, `number`, or some built-in type.

Reply:
Sure thing, but if the domain become much more complex, it's really hard to maintain each basic types.

e.g. If we change the `UserId` from type `number` to `string`, we need to find out every interfaces or types  that contains `UserId` and change it.

I treat these types more like our custom primitive types.